src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,236 @@
+from triage.component.results_schema import upgrade_db, Experiment, ExperimentModel 
+from triage.component.architect.entity_date_table_generators import EntityDateTableGenerator, DEFAULT_ACTIVE_STATE
+from triage.component.architect.features import FeatureGenerator
+from triage.component.architect.feature_group_creator import FeatureGroup
+from triage.component.architect.builders import MatrixBuilder
+from triage.component.architect.planner import Planner
+from triage.component.timechop import Timechop
+from triage.component.catwalk.predictors import Predictor
+from triage.component.catwalk.utils import filename_friendly_hash
+from triage.util.conf import convert_str_to_relativedelta, dt_from_str
+from triage.util.db import scoped_session
+from sqlalchemy import select
+
+from collections import OrderedDict
+import json
+import re
+
+import verboselogs, logging
+logger = verboselogs.VerboseLogger(__name__)
+
+
+def experiment_config_from_model_id(db_engine, model_id):
+    """Get original experiment config from model_id 
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+
+    Returns: (dict) experiment config
+    """
+    get_experiment_query = '''select experiments.config
+    from triage_metadata.experiments
+    join triage_metadata.experiment_models using (experiment_hash)
+    join triage_metadata.models using (model_hash)
+    where model_id = %s
+    '''
+    (config,) = db_engine.execute(get_experiment_query, model_id).first()
+    return config
+    
+
+def train_matrix_info_from_model_id(db_engine, model_id):
+    """Get original train matrix information from model_id 
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+
+    Returns: (str, dict) matrix uuid and matrix metadata
+    """
+    get_train_matrix_query = """
+        select matrix_uuid, matrices.matrix_metadata
+        from triage_metadata.matrices
+        join triage_metadata.models on (models.train_matrix_uuid = matrices.matrix_uuid)
+        where model_id = %s
+    """
+    return db_engine.execute(get_train_matrix_query, model_id).first()
+
+
+def get_feature_names(aggregation, matrix_metadata):
+    """Returns a feature group name and a list of feature names from a SpacetimeAggregation object"""
+    feature_prefix = aggregation.prefix
+    logger.spam("Feature prefix = %s", feature_prefix)
+    feature_group = aggregation.get_table_name(imputed=True).split('.')[1].replace('"', '')
+    logger.spam("Feature group = %s", feature_group)
+    feature_names_in_group = [f for f in matrix_metadata['feature_names'] if re.match(f'\\A{feature_prefix}', f)]
```

Comment:
If you have one feature prefix `foo` and one `foobar`, won't trying to match on `foo` here end up grabbing both sets of features? Maybe the regex match here should include the `_`?

Reply:
Looks like you're right. We can make that change.

However it looks like we allow underscores in prefixes, which could also allow a screwup if you have `foo` and `foo_bar` as prefixes.

This is another place where having a feature metadata table would be great, because the feature group/prefix data could be directly encoded there. In the meantime, I'm not sure that allowing underscores in prefix names is wise. Banning them might also encourage short feature group prefixes which would make it less likely they'll hit the character max.

Reply:
Huh... I thought we'd run into some similar issue here with joco and changed something in the config validation as a result. Or am I thinking of capitalization? @ecsalomon -- do you remember better?

Maybe another simple option is to take the longest feature prefix that matches since there will generally only be a handful you have to check? Though I agree that feature metadata is definitely the right longer-term solution!